backend/backend/core/agents.py

```diff
@@ -1,22 +1,27 @@
 from langchain_core.prompts import ChatPromptTemplate
+from langgraph.errors import GraphRecursionError
 from langgraph.prebuilt import create_react_agent
 
 from utils.logger import logger
 from .models import ModelRegistry
 from .tools import ToolRegistry
+from .chains import ChainRegistry
 
 
 class EventsAgent(object):
     def __init__(self, model, search_tool):
         super().__init__()
         prompt = ChatPromptTemplate.from_template(
-            "Generate a list of events and short descriptions happening in {location} on {date}")
+            "Generate a list of events and short descriptions happening in {location} on {date}"
+        )
         agent = create_react_agent(model, [search_tool])
         self.chain = prompt | agent
 
     async def ainvoke(self, location, date):
         logger.info(
-            f"EventsAgent: Getting events for location: {location} and date: {date}")
+            f"EventsAgent: Getting events for location: {
```

Comment:
The line number in the logger.info function call is not necessary and can be removed.